app/controllers/data_api/v1/charges_controller.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+module DataApi
+  module V1
+    class ChargesController < DataApi::BaseController
+      include PremiumFeatureOnly
+
+      def bulk_forecasted_usage_amount
+        charges_data = params[:charges] || []
+
+        if charges_data.empty?
+          render json: {error: "No charges provided"}, status: :bad_request
+          return
+        end
+
+        charge_ids = charges_data.map { |cd| cd[:charge_id] }.compact.uniq
```

Comment:
A lor of logic is happening in this controller. Usually we tend to delegate the logic to a service class.

It helps with the single responsibility principle, and make it easier to format the controller response by relying on existing response methods defined in `render_error_response`